Work to solve https://rt.cpan.org/Ticket/Display.html?id=66469

pplu opened the pull request:
Hi!

This is the result of the work we commented over IRC.

pplu committed:
Be able to override the autodetected repo name

Use this capability in Git::CollectionOfRepositories::FromDirectoryRecursive

pplu committed:
Make get_repository expect already instanced repo

Refactor subclasses to have the _get_repo_from_name as the mandatory
method.

pplu committed:
Ammend test of DirectoryRecursive

Now the names of the repos that DirectoryRecursive returns are relative
paths.

Also add two repos with the same name, in different directories to the
test suite.

pplu committed:
More DirectoryRecursive testing

Make sure that two repos with the same name are retrievable, and are
different.

pplu committed:
More tests changed when a new repo was added

Fix model_collectionofrepos.t tests

pplu committed:
Fixed tree view for repos with / in name

The "/" in the name of the repo was not escaped. Use uri_for_action to
generate the link to the fragments URI. That implies deleting the
"around" uri_for_action in Gitalist.pm. That seems to be OK because
tests still pass. Also, didn't find any invocation to uri_for_action
with a /fragments path.

broquaint commented:
Thanks for this, pplu, I've had a quick scan of the diff and want to poke about the code a bit before commenting on the diff. It looks like good work, thanks for the contribution!

pplu committed:
Force user to select the radios in the diff

500 Error when a version was not selected caused because the
interface passes "undefined" as an sha1 to diff.

If the diff screen only had one entry, only one radio button would be selected, so
clicking the "Compare" link would trigger the error.

broquaint commented:
I've now merged your changes\* to macquart and subsequently master which has just gone out on the CPAN at version 0.003001 (on all good CPAN mirrors near you soon).

Thanks for the great contribution!
- Bar the comparison JS, will roll that into another dev branch.
